internal/graph/graph.go

```diff
@@ -385,3 +385,43 @@ func (g *RelationshipGraph) getRelationshipEdgesWithTargetRewrite(
 		panic("unexpected userset rewrite encountered")
 	}
 }
+
+// NewLayeredCheckResolver constructs a CheckResolver that is composed of various CheckResolver layers.
+// Specifically, it constructs a CheckResolver with the following composition:
+//
+//	CycleDetectionCheckResolver  <-----|
+//		CachedCheckResolver              |
+//			LocalChecker                   |
+//				CycleDetectionCheckResolver -|
+//
+// The returned CheckResolverCloser should be used to close all resolvers involved in the
+// composition after you are done with the CheckResolver.
+func NewLayeredCheckResolver(
+	localResolverOpts []LocalCheckerOption,
+	cacheEnabled bool,
+	cachedResolverOpts []CachedCheckResolverOpt,
```

Comment:
nit: if `cacheEnabled` is false, having to pass an empty `cachedResolverOpts` is... weird? I recommend making `cacheEnabled` part of `CachedCheckResolverOpt`

Reply:
`cacheEnabled` controls not only if the CachedCheckResolver gets constructed but also if if re-wire the other delegates based on if it is enabled. If we refactor `WithCacheEnabled` as a `CachedCheckResolverOpt` we lose the ability to make that determination.

With that in mind, how would you suggest the composition of the code look like to behave the same but remove the stutter?